File: src/main/java/com/paragon/client/shader/shaders/DiamondsShader.java
package com.paragon.client.shader.shaders;

import com.paragon.client.shader.Shader;

import java.awt.*;

import static org.lwjgl.opengl.GL20.*;

public class DiamondsShader extends Shader {

    private Color colour;
    private float size;
    private float spacing;

    public DiamondsShader() {
        super("/assets/paragon/glsl/shaders/diamonds.frag");
    }

    public void setColor(Color colour) {
        this.colour = colour;
    }

    public void setSize(float size) {
        this.size = size;
    }

    public void setSpacing(float spacing) {
        this.spacing = spacing;
    }

    @Override
    public void setupUniforms() {
        setupUniform("texture");
        setupUniform("texelSize");
        setupUniform("resolution");
        setupUniform("time");

        setupUniform("size");
        setupUniform("spacing");
        setupUniform("colour");
    }

    @Override
    public void updateUniforms() {
        glUniform1i(getUniform("texture"), 0);
        glUniform2f(getUniform("texelSize"), 1F / mc.displayWidth, 1F / mc.displayHeight);
        glUniform2f(getUniform("resolution"), mc.displayWidth, mc.displayHeight);
        glUniform1f(getUniform("time"), (float) getTime());

        glUniform4f(getUniform("colour"), colour.getRed() / 255f, colour.getGreen() / 255f, colour.getBlue() / 255f, colour.getAlpha() / 255f);
        glUniform1f(getUniform("size"), size);
        glUniform1f(getUniform("spacing"), spacing);
    }
}

File: src/main/kotlin/com/paragon/util/render/shader/shaders/DiamondsShader.kt
package com.paragon.util.render.shader.shaders

import com.paragon.util.render.shader.Shader
import org.lwjgl.opengl.GL20.*
import java.awt.Color

class DiamondsShader : Shader("/assets/paragon/glsl/shaders/diamonds.frag") {

    private var colour: Color? = null
    private var size = 0f
    private var spacing = 0f

    fun setColor(colour: Color?) {
        this.colour = colour
    }

    fun setSize(size: Float) {
        this.size = size
    }

    fun setSpacing(spacing: Float) {
        this.spacing = spacing
    }

    override fun setupUniforms() {
        setupUniform("texture")
        setupUniform("time")
        setupUniform("size")
        setupUniform("spacing")
        setupUniform("colour")
    }

    override fun updateUniforms() {
        glUniform1i(getUniform("texture"), 0)
        glUniform1f(getUniform("time"), time.toFloat())
        glUniform4f(getUniform("colour"), colour!!.red / 255f, colour!!.green / 255f, colour!!.blue / 255f, colour!!.alpha / 255f)
        glUniform1f(getUniform("size"), size)
        glUniform1f(getUniform("spacing"), spacing)
    }

}